src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.SymbolKeyReader.cs

```diff
@@ -417,9 +499,10 @@ public void PopMethod(IMethodSymbol? method)
 
             #region Symbols
 
-            public SymbolKeyResolution ReadSymbolKey(out string? failureReason)
+            public SymbolKeyResolution ReadSymbolKey(ITypeSymbol? contextualType, out string? failureReason)
```

Comment:
i intentinoally updated this method, forcing all callers to ahve to consider what to pass in here.  it was too easy to forget to set the contextual type in the caller